### PR TITLE
feat(web): templates store use feature flag

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

Templates Store uses the feature flag.

### Why was this change needed?

Templates Store PRD.

### Other information (Screenshots)

https://github.com/novuhq/novu/assets/2607232/cbe7ea9c-b612-4fea-b05c-069d89e60217

### CODE REVIEW

File: apps/api/src/app/blueprint/usecases/get-grouped-blueprints/consts.ts [hunk 1]
```diff
@@ -66,8 +66,7 @@ export const POPULAR_TEMPLATES_GROUPED = {
     {
       _id: '1',
       name: ':fa-regular fa-message: Comments',
-      description:
-        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae. Ipsam quasi fugiat commodi adipisci eligendi necessitatibus cumque aliquam, dicta natus cupiditate suscipit voluptatum rerum debitis. Ipsum!',
+      description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae.',
```
Comment (LetItRock) [inline apps/api/src/app/blueprint/usecases/get-grouped-blueprints/consts.ts:69]: the description should be less or equal 100 characters

File: apps/api/src/app/blueprint/usecases/get-grouped-blueprints/consts.ts [hunk 2]
```diff
@@ -76,12 +75,21 @@ export const POPULAR_TEMPLATES_GROUPED = {
...
+      active: true,
+      draft: false,
+      critical: false,
+      preferenceSettings: {
+        chat: true,
+        email: true,
+        in_app: true,
+        push: true,
+        sms: true,
+      },
```
Comment (LetItRock) [inline apps/api/src/app/blueprint/usecases/get-grouped-blueprints/consts.ts:87]: missing fields

File: apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts [hunk 3]
```diff
@@ -51,13 +51,22 @@ export class GetGroupedBlueprints {
...
-      return (storedBlueprint ? storedBlueprint : localBlueprint) as INotificationTemplate;
+      return (
+        storedBlueprint
+          ? storedBlueprint
+          : {
+              ...localBlueprint,
+              _notificationGroupId: firstBlueprint._notificationGroupId,
+              notificationGroup: firstBlueprint.notificationGroup,
+            }
+      ) as INotificationTemplate;
```
Comment (LetItRock) [inline apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts:69]: Basically, the popular blueprints are hardcoded constants that don't have the notification group and this raises an API error when creating the workflow from it on the FE... 

I'm not sure what is the best way to bypass this, but here I'm assigning the first blueprint notification group...

@djabarovgeorge cc

File: apps/web/src/pages/templates/TemplatesListNoDataOld.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import styled from '@emotion/styled';
```
Comment (LetItRock) [inline apps/web/src/pages/templates/TemplatesListNoDataOld.tsx:1]: reverted the old view shown when there is no workflows, where we have two tiles: "Create Workflow" and "Try the Digest playground"

File: apps/web/src/pages/templates/TemplatesListPage.tsx [hunk 5]
```diff
@@ -142,20 +152,61 @@ function NotificationList() {
...
+                blueprints={popular?.blueprints}
+                isLoading={areBlueprintsLoading}
+                isCreating={isCreatingTemplateFromBlueprint}
+                allTemplatesDisabled={areBlueprintsLoading || !hasGroups}
+                onBlankWorkflowClick={() => handleRedirectToCreateTemplate(false)}
+                onTemplateClick={handleOnBlueprintClick}
+                onAllTemplatesClick={openModal}
+              />
+            </When>
+          </>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/TemplatesListPage.tsx:208]: when templates store feature flag enabled

File: apps/web/src/pages/templates/TemplatesListPage.tsx [hunk 6]
```diff
@@ -168,17 +219,13 @@ function NotificationList() {
...
-            onBlankWorkflowClick={() => handleRedirectToCreateTemplate(false)}
-            onTemplateClick={handleOnBlueprintClick}
-            onAllTemplatesClick={openModal}
+            noDataPlaceholder={
+              <TemplatesListNoDataOld
+                onCreateClick={() => handleRedirectToCreateTemplate(false)}
+                onTryDigestClick={handleCreateDigestDemoWorkflow}
+                tryDigestDisabled={isTryDigestDisabled}
+              />
+            }
```
Comment (LetItRock) [inline apps/web/src/pages/templates/TemplatesListPage.tsx:228]: when the templates store feature flag is disabled

File: apps/api/src/app/blueprint/usecases/get-grouped-blueprints/consts.ts [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
+import {
+  DigestTypeEnum,
+  DigestUnitEnum,
+  INotificationTemplate,
+  INotificationTemplateStep,
+  StepTypeEnum,
+} from '@novu/shared';
 
-const STEPS = [
+const STEPS: INotificationTemplateStep[] = [
```
Comment (LetItRock) [inline apps/api/src/app/blueprint/usecases/get-grouped-blueprints/consts.ts:11]: I've noticed that after clicking on "Use template" button the Workflow Editor shows some errors on the steps, so I've fixed them here

File: apps/web/src/pages/templates/components/templates-store/TemplatesStoreModal.tsx [hunk 8]
```diff
@@ -102,6 +103,19 @@ export const TemplatesStoreModal = ({ general, isOpened, onClose }: ITemplatesSt
...
+              {group.blueprints.map((template) => {
+                return (
+                  <TemplateItem key={template.name} onClick={() => handleTemplateClick(template)}>
+                    <FontAwesomeIcon icon={template.iconName} />
+                    <span>{template.name}</span>
+                  </TemplateItem>
+                );
+              })}
+            </TemplatesGroup>
+          ))}
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/templates-store/TemplatesStoreModal.tsx:118]: forgot to show the popular blueprints the the Templates Store modal, they will be rendered at the top

File: apps/web/src/pages/templates/hooks/useTemplatesStoreModal.tsx [hunk 9]
```diff
@@ -1,23 +1,34 @@
...
+}: {
+  general?: IBlueprintsGrouped[];
+  popular?: IBlueprintsGrouped;
+}) => {
+  const [opened, { open, close }] = useDisclosure(false);
+  const isTemplateStoreEnabled = useIsTemplateStoreEnabled();
   const hasGroups = general && general.length > 0;
+  const hasPopular = !!(popular && popular?.blueprints.length > 0);
+  const hasGroupsOrPopular = hasGroups || hasPopular;
+  const isOpened = opened && hasGroupsOrPopular && isTemplateStoreEnabled;
```
Comment (p-fernandez) [inline apps/web/src/pages/templates/hooks/useTemplatesStoreModal.tsx:21]: Looks like all this information could be living in its own hook for tidiness, though I acknowledge the reusability might be non existent.

**Review by p-fernandez — APPROVED** [review]
Smooth 🍹 
🌟

File: apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts (hunk 3 above)
Comment (djabarovgeorge) [reply apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts:69]: Due to our decision last week (don't use hardcoded data directly on the client) the only thing that we will need is the ID, so we won't have this issue.

**Review by djabarovgeorge — APPROVED** [review]
🔥

**Pull request merged**